README.md

```diff
@@ -1,32 +1,120 @@
-# AWS %description% Terraform module
+# Amazon Kinesis to Fluent log transfer Terraform module
 
-Terraform module for %long_description%
+Terraform module and Lambda for sending JSON log records from Kinesis Data Streams to a Fluent endpoint.
 
 ![terraform v0.11.x](https://img.shields.io/badge/terraform-v0.11.x-brightgreen.svg)
 
-## Usage
+## Prerequisites
+1. Records in Kinesis stream must be valid JSON data. Non-JSON data will be saved with `unknown` prefix.
+    1. gzipped JSON, [CloudWatch Logs subscription filters log format](https://docs.aws.amazon.com/ja_jp/AmazonCloudWatch/latest/logs/SubscriptionFilters.html) are supported.
+    2. Logs without either of necessary keys listed below will be saved as `unknown` as well.
+2. JSON data must have the following keys (key names are modifiable via variables):
+    1. `log_type`: Log type identifier. Failed logs  will be saved by this key: `%log_type%/YYYY-MM/DD/`. 
+    2. `log_id`: Any unique identifier. Used to avoid file overwrites on S3. Also is useful to search for a specific log record.
+    3. `time`: Any timestamp supported by [dateutil.parser.parse](https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.parse). ISO8601 with milli/microseconds recommended. This timestamp is used when emitting logs with fluentd protocol; message timestamp will be set to this timestamp. 
 
+## Usage
 ```HCL
-resource "aws_s3_bucket" "bucket" {
-    bucket = "test"
+data "aws_vpc" "vpc" {
+  filter {
+    name   = "tag:Name"
+    values = ["main"]
+  }
 }
+
+data "aws_subnet_ids" "private" {
+  vpc_id = "${data.aws_vpc.vpc.id}"
+  
+  filter {
+    name   = "tag:Role"
+    values = ["private"]
+  }
+}
+
+resource "aws_security_group" "all_outbound" {
+  name   = "kinesis_to_fluent-sg"
+  vpc_id = "${data.aws_vpc.vpc.id}"
+  
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "aws_kinesis_stream" "stream" {
+  name             = "stream"
+  shard_count      = 1
+  retention_period = 24
+}
+
+module "fluent" {
+  source  = "baikonur-oss/lambda-kinesis-to-s3/aws"
+  version = "1.0.0"
+
+  lambda_package_url = "https://github.com/baikonur-oss/terraform-aws-lambda-kinesis-fluent/releases/download/v1.0.0/lambda_package.zip"
+  name               = "kinesis_to_fluent"
+
+  batch_size         = 100
+  kinesis_stream_arn = "${aws_kinesis_stream.stream.arn}"
+  fluent_tag         = "dev"
+  fluent_target_url  = "10.20.30.40"
```

Comment:
IMO, `fluent_target_hostname` is more suitable for representing the expected value from the user.

Reply:
Maybe `fluent_sink_endpoint` would be better? This could be a hostname or could be an IP address.

Reply:
I see, it sounds reasonable.